src/interfaces/InjectedSenderWallet.ts

```diff
@@ -101,16 +91,15 @@ export interface RequestSignTransactionsParams {
 }
 
 interface InjectedSenderWallet {
-  init: (params: InitParams) => Promise<InitResponse>;
+  isSender: string,
   getAccountId: () => string;
   getRpc: () => Promise<GetRpcResponse>;
   requestSignIn: (
     params: RequestSignInParams
   ) => Promise<RequestSignInResponse>;
-  signOut: () => Promise<SignOutResponse>;
+  signOut: () => SignOutResponse;
   isSignedIn: () => boolean;
-  onAccountChanged: (callback: AccountChangedCallback) => void;
-  onRpcChanged: (callback: RpcChangedCallback) => void;
+  on: (event: string, callback: Callback) => void;
```

Comment:
To address the `any` issue (this project forbids them), we should look to create an event map to ensure callback methods have the correct types:

```ts
export interface SenderWalletEvents {
  signIn: () => void;
  signOut: () => void;
  accountChanged: (changedAccountId: string) => void;
  rpcChanged: (response: RpcChangedResponse) => void;
}

on: <Event extends keyof SenderWalletEvents>(
    event: Event,
    callback: SenderWalletEvents[Event]
  ) => void;
```

Reply:
great. will add the event map. .